sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCSuite.scala

```diff
@@ -44,7 +45,8 @@ import org.apache.spark.sql.test.SharedSparkSession
 import org.apache.spark.sql.types._
 import org.apache.spark.util.Utils
 
-class JDBCSuite extends QueryTest with SharedSparkSession {
```

Comment:
not related to this PR: AQE is on by default, how did this test suite still run with AQE off?

Reply:
In this PR, `JDBCSuite` runs without AQE enabled as now the suite inherits `DisableAdaptiveExecutionSuite`. In comparison, `JDBCWithAQESuite` inherits `EnableAdaptiveExecutionSuite`

Reply:
I mean before this PR, the test code didn't consider AQE but still passed. This is weird because AQE is on by default.

Reply:
I see your point. I revisited this, and yes AQE is enabled. However, AQE was not actually not applied in the execution plan as `shouldApplyAQE()` returns `false`. In that sense, this PR description should have been `AQE applied` instead of `AQE enabled`.

Reply:
does it mean this PR is testing something that will never happen in production?

Reply:
It is actually opposite. This PR is adding the test for when user enables `spark.sql.adaptive.forceApply` that can happen in production.

Reply:
In that sense, I could have not to add `DisableAdaptiveExecutionSuite` in the base suite, so that it is closer to the default.

Reply:
`spark.sql.adaptive.forceApply` is designed for testing. Do you have use cases to turn it on in production? If use cases are valid, we should make it a public conf (instead of internal).

Reply:
Got it. I missed that part. Let me send reverting PRs for those 3 test suites

Reply:
thanks for your understanding!